packages/client/src/helpers/RNSpeechDetector.ts

```diff
@@ -0,0 +1,114 @@
+import { BaseStats } from '../stats/types';
+import { SoundStateChangeHandler } from './sound-detector';
+
+/**
+ * Flatten the stats report into an array of stats objects.
+ *
+ * @param report the report to flatten.
+ */
+const flatten = (report: RTCStatsReport) => {
+  const stats: RTCStats[] = [];
+  report.forEach((s) => {
+    stats.push(s);
+  });
+  return stats;
+};
+
+const AUDIO_LEVEL_THRESHOLD = 0.2;
+
+export class RNSpeechDetector {
+  private pc1 = new RTCPeerConnection({});
+  private pc2 = new RTCPeerConnection({});
+  private intervalId: NodeJS.Timer | undefined;
+
+  /**
+   * Starts the speech detection.
+   */
+  public async start() {
+    try {
+      if (this.pc1 && this.pc2) {
```

Comment:
`pc1` and `pc2` are always defined now. Perhaps this `if` condition can be removed.
Also `this.pc1?.` -> `this.pc1.`

Reply:
Done